src/main.cpp

```diff
@@ -5936,7 +5936,7 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
     else if (strCommand == NetMsgType::XPEDITEDBLK)
     {
         // ignore the expedited message unless we are at the chain tip...
-        if (!fImporting && !fReindex && !IsInitialBlockDownload() && IsThinBlocksEnabled())
```

Comment:
This is actually not a redundant check and is needed in order to turn thinblocks off completely...so here we're checking if *we* have thinblocks turn off and if so then we don't accept any expedited xthins.  The other check below tell us if the remote peer has it turned on or not.